lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -239,111 +214,114 @@ export default function LMSFilePicker({
+        continueAction = { type: 'reload' };
+      } else {
+        continueAction = {
+          type: 'select',
+          label: 'Select',
+          disabled: selectedFile === null,
+        };
+      }
+      break;
+    case 'authorizing':
+      continueAction = {
+        type: 'authorize',
+        label: dialogState.isRetry ? 'Try again' : 'Authorize',
+      };
+      break;
+    case 'error':
+      continueAction = {
+        type: 'authorize',
+        label: 'Try again',
+      };
+      break;
+  }
+
+  let continueButton;
+  switch (continueAction.type) {
+    case 'authorize':
+      continueButton = (
         <AuthButton
           authURL={/** @type {string} */ (listFilesApi.authUrl)}
           authToken={authToken}
-          label="Try again"
-          onAuthComplete={fetchFiles}
-          data-testid="try-again"
-        />
-      ),
-      warningOrError: (
-        <ErrorDisplay
-          message="There was a problem fetching files"
-          error={/** @type {Error} */ (dialogState.error)}
+          label={continueAction.label}
+          onAuthComplete={() => fetchFiles(true /* reload */)}
         />
-      ),
-    },
-    reload: {
-      continueButton: (
+      );
+      break;
+    case 'reload':
+      continueButton = (
         <LabeledButton
-          disabled={dialogState.state === 'reloading'}
-          onClick={fetchFiles}
           variant="primary"
+          onClick={() => fetchFiles(true /* reload */)}
           data-testid="reload"
         >
           Reload
         </LabeledButton>
-      ),
-      warningOrError: null,
-    },
-  };
-
-  const { continueButton, warningOrError } =
-    options[dialogState.continueAction];
+      );
+      break;
+    case 'select':
+      continueButton = (
+        <LabeledButton
+          variant="primary"
+          disabled={continueAction.disabled}
+          onClick={() => confirmSelectedItem()}
+          data-testid="select"
+        >
+          {continueAction.label}
+        </LabeledButton>
+      );
+      break;
+  }
 
-  // During the first fetch and load, we don't know yet whether we'll have
-  // success fetching files, or whether we'll need auth'z or hit another
-  // error. To avoid hopscotching between modals, show only
-  // a full-screen spinner during this state.
-  const isLoading = dialogState.state === 'fetching';
-  // During subsequent file/folder fetches, represent the loading state within
-  // the file list UI (do not show a full-screen spinner)
-  const isReloading = dialogState.state === 'reloading';
-  // Render the file list and breadcrumbs UI in certain states. This also
-  // requires more real estate in the <Modal>, and applies an appropriate CSS
-  // class
-  const withFileUI = ['reloading', 'fetched'].includes(dialogState.state);
+  const withFileUI = ['fetching', 'fetched'].includes(dialogState.state);
 
-  if (isLoading) {
+  if (dialogState.state === 'fetching' && initialFetch) {
```

Comment:
#2736 has this empty-ish render return considerably higher up (as a bunch of render logic is irrelevant to it), but I find that it can be easily lost that way (i.e. I prefer to have all render returns relatively near each other). I had a specific experience when working in this component about a week ago wherein I lost track of the empty-render branch for a while and it caused me confusion. Thoughts?